src/Behaviors/Meshes/baseSixDofDragBehavior.ts

```diff
@@ -223,20 +227,30 @@ export class BaseSixDofDragBehavior implements Behavior<Mesh> {
                     ) {
                         pointerInfo.pickInfo.ray.origin.copyFrom(this._pointerCamera!.globalPosition);
                     }
-
                     this._dragging = true;
+
                     virtualMeshesInfo.lastOriginPosition.copyFrom(pointerInfo.pickInfo.ray.origin);
 
                     // Set position and orientation of the controller
                     virtualMeshesInfo.originMesh.position.copyFrom(pointerInfo.pickInfo.ray.origin);
-                    virtualMeshesInfo.originMesh.lookAt(pointerInfo.pickInfo.ray.origin.add(pointerInfo.pickInfo.ray.direction));
+                    if (pointerInfo.pickInfo.originMesh) {
+                        // Near interaction
+                        virtualMeshesInfo.originMesh.lookAt(this._pointerCamera!.globalPosition);

```

Comment:
Does this mean that when dragging objects in XR with near interaction, they'll be oriented to face the user/camera? Or does the virtual mesh not relate to the real mesh in that way?

Reply:
`baseSixDofDragBehavior` only moves around virtual meshes related to pointer positions/pick, then the mesh position/orientation is set in the child class `SixDofDragBehavior`. This separation was made to enable other kind of pointer interactions, like 2 pointer interactions, or the plane constrained drag for slate gizmo's handles.

So yes, even though the virtual mesh looks at camera, final mesh orientation is indeed different.